src/Carousel/Carousel.xml

```diff
@@ -0,0 +1,194 @@
+                <caption>Entity</caption>
+                <category>Data source</category>
+                <description>The entity which should be displayed in the carousel</description>
+            </property>
+            <property key="imageSource" type="enumeration" defaultValue="microflow">
+                <caption>Source</caption>
+                <category>Data source</category>
+                <description>Type of source</description>                
+                <enumerationValues>
+                    <enumerationValue key="xpath">XPath</enumerationValue>
+                    <enumerationValue key="microflow">Microflow</enumerationValue>
+                    <enumerationValue key="static">Static</enumerationValue>
+                </enumerationValues> 
+            </property>
+            <property key="entityConstraint" type="entityConstraint" entityProperty="imageEntity" required="false">
+			    <caption>Constraint</caption>
+			    <category>Source - XPath</category>
+			    <description>Constraint on the carousel entity</description>
+	        </property>
+            <property key="dataSourceMicroflow" type="microflow" required="false">
+                <caption>Data source microflow </caption>
+                <category>Source - microflow</category>
+                <description>Returns images to display in the carousel</description>
+                <returnType type="Object" entityProperty="imageEntity" isList="true"/>
+	        </property>
+            <property key="staticImageCollection" type="object" isList="true" required="false">
+                <caption>Images to display</caption>
+                <category>Source - static</category>
+                <description>Images to display in the carousel</description> 
+                <properties>
+                    <property key="caption" type="string" required="false">
+                        <caption>Caption</caption>
+                        <category>Data source</category>
+                        <description>Caption to display on the image</description>
+                    </property>
+                    <property key="description" type="string" required="false">
+                        <caption>Description</caption>
+                        <category>Data source</category>
+                        <description>Description to display on the image</description>
+                    </property>
+                    <property key="pictureUrl" type="image" required="true">
+                        <caption>Image</caption>
+                        <category>Data source</category>
+                        <description>Image to add to the carousel</description>
+                    </property>
+
+                    <property key="onClickEvent" type="enumeration" defaultValue="none">
+                        <caption>On click</caption>
+                        <category>Events</category>
+                        <description>Event to execute</description>
```

Comment:
The person configuring the widget isn't "executing an event". He or she is configuring what to do when an end-user is clicking or touching on an image.